src/lib/app/streams.ts

```diff
@@ -5,18 +5,27 @@ import type {
     RequestSubscriptionResponse,
     Subscription,
 } from '@iota/is-client'
-import { AccessRights, ChannelInfo } from '@iota/is-client'
+import { AccessRights, type ChannelInfo } from '@iota/is-client'
 import type { Writable } from 'svelte/store'
 import { get, writable } from 'svelte/store'
 import { authenticationData, channelClient, isAuthenticated } from './base'
 import { DEFAULT_SDK_CLIENT_REQUEST_LIMIT } from './constants/base'
-import { FEED_INTERVAL_MS } from './constants/streams'
+import { DEFAULT_AUTHOR_FILTER_STATE, FEED_INTERVAL_MS } from './constants/streams'
 import { showNotification } from './notification'
 import { NotificationType } from './types/notification'
-import { SubscriptionState } from './types/streams'
+import { SubscriptionState, type StreamsFilter } from './types/streams'
 
 export const selectedChannelPageIndex: Writable<number> = writable(1)
 export const channelSearchQuery: Writable<string> = writable('')
+export const channelFilterOptions: Writable<StreamsFilter> = writable({
+    limitFilter: {
```

Comment:
what is this limitFilter? and why one has a state and value but the other only a state? What is the state saying?

Reply:
The limitFilter is the query limit. Both have a default state. The value for the authorFilter can only be set in the component itself because the `authenticatedUserDID` is not available when streams.ts is loaded. I added comments to the Filter type in shared.ts for clarification.

Reply:
Sorry this is not straight forward to understand since a checkbox can only be true or false. Please remove state and value it shall just be a boolean.

Reply:
For the limitFilter you are using the value to set the filter but for authorFilter you are using the state to define if it is checked. For me it seems like state can be completely removed here but only use value.

Reply:
I removed value and just added state. It works best this way around.